### PR TITLE
[OpenMP] Clang Codegen Interop : Accept multiple init

### PR DESCRIPTION
**Pull request by SunilKuravinakop**
Modifying clang/lib/CodeGen/CGStmtOpenMP.cpp to accept multiple `init` clauses with `interop` directive.

### CODE REVIEW

File: clang/lib/CodeGen/CGStmtOpenMP.cpp [hunk 1]
```diff
@@ -7023,19 +7023,25 @@ void CodeGenFunction::EmitOMPInteropDirective(const OMPInteropDirective &S) {
...
-    llvm::omp::OMPInteropType InteropType = llvm::omp::OMPInteropType::Unknown;
-    if (C->getIsTarget()) {
-      InteropType = llvm::omp::OMPInteropType::Target;
-    } else {
-      assert(C->getIsTargetSync() && "Expected interop-type target/targetsync");
-      InteropType = llvm::omp::OMPInteropType::TargetSync;
+  auto It = S.getClausesOfKind<OMPInitClause>();
+  if (!It.empty()) {
+    // Look at the multiple init clauses
+    for (auto C : It) {
```
Comment (alexey-bataev) [inline clang/lib/CodeGen/CGStmtOpenMP.cpp:7029]: Expand 'auto' here

Comment (SunilKuravinakop) [reply clang/lib/CodeGen/CGStmtOpenMP.cpp:7029]: For the Iterator variable `It` `auto It` is preferable rather than 
`llvm::iterator_range<clang::OMPExecutableDirective::specific_clause_iterator<clang::OMPInitClause> > It`

Comment (alexey-bataev) [reply clang/lib/CodeGen/CGStmtOpenMP.cpp:7029]: C is just OMPClause * here, not an iterator

Comment (alexey-bataev) [reply clang/lib/CodeGen/CGStmtOpenMP.cpp:7029]: I asking about auto for C, not for It.

**Comment by SunilKuravinakop** [comment]
Alexey, 
Can you please review my changes?

File: clang/lib/CodeGen/CGStmtOpenMP.cpp [hunk 2]
```diff
@@ -7023,31 +7023,47 @@ void CodeGenFunction::EmitOMPInteropDirective(const OMPInteropDirective &S) {
...
+  auto ItOMPUseClause = S.getClausesOfKind<OMPUseClause>();
+  if (!ItOMPUseClause.empty()) {
+    // Look at the multiple use clauses
+    for (const OMPUseClause *C : ItOMPUseClause) {
+      llvm::Value *InteropvarPtr =
+          EmitLValue(C->getInteropVar()).getPointer(*this);
+      OMPBuilder.createOMPInteropUse(Builder, InteropvarPtr, Device,
+                                     NumDependences, DependenceList,
+                                     Data.HasNowaitClause);
     }
```
Comment (alexey-bataev) [inline clang/lib/CodeGen/CGStmtOpenMP.cpp:7066]: Split it to a separate patch

**Review by alexey-bataev — APPROVED** [review]
LG
